sound/soc/sof/intel/cnl.c

```diff
@@ -230,6 +230,8 @@ struct snd_sof_dsp_ops sof_cnl_ops = {
 	.pcm_hw_params	= hda_dsp_pcm_hw_params,
 	.pcm_trigger	= hda_dsp_pcm_trigger,
 
+	.dai_config	= hda_dai_config,
```

Comment:
is this really specific to cnl?
How would you handle the case of a platform which can work in either I2S or HDaudio modes, with a run-time selection done?
and does this really belong in the ops or should it be a hook private to HDaudio cases?

Reply:
yes, APL and ICL should be included.

I will add a macro ifdef _HDA_ to control it

Reply:
Humm, not sure. Even if you have HDA configured, you may have DMICs. So not all DAIs will support this callback. and since you invoke the callback from HDaudio code only, this might be better handled from a HDaudio-specific ops or helper.

Reply:
yes, I made  a change